django/conf/__init__.py

```diff
@@ -109,6 +110,29 @@ def configure(self, default_settings=global_settings, **options):
             setattr(holder, name, value)
         self._wrapped = holder
 
+    @staticmethod
+    def _add_script_prefix(value):
+        """
+        Prefix header value SCRIPT_NAME to ``value`` under certain circumstances.
+
+        Useful for when the app is being served at a subpath and manually prefixing
+        that subpath to STATIC_URL and STATIC_ROOT in settings is inconvenient
+        (i.e., a deployment concern).
+        """
+        try:
+            URLValidator()(value)
+            return value
+        except (ValidationError, AttributeError):
+            pass
+        from django.urls import get_script_prefix  # circular dependency if top-level import
+        prefix = get_script_prefix()
+        if not prefix or prefix == '/':
+            return value
+        # try to detect if the prefix is already prefixed appropriately.
+        if value.startswith(prefix.rstrip('/')):
```

Comment:
As said in the previous PR already, I think we should drop this

Reply:
I quickly looked thru previous PR comments, but didn't really find solid background reasons. Could you please explain why?

Without this we will for sure break a lot of websites which use hardcoded subfolder already. This tries to mitigate this at least a little bit. If you don't mind explicit breaking of backward compatibility then for sure I can remove this part.

Reply:
I think the conclusion on the ticket/old PR was that prefixing would only happen for relative paths, which is not something that is currently allowed. Looking through the docs below, I see that this isn't really clear currently. From what I can see this PR currently still prefixes even if the url starts with a /, this is not something we should do.

The final behavior that we agreed on was documented in https://github.com/django/django/pull/10724#issuecomment-445946077 -- why did you revert that behavior again?

Reply:
Because I do not agree with not prefixing absolute path. This change does not make sense if it does not prefix absolute path. I'm not even sure that there are use cases where one uses relative `STATIC_URL` - because it will not work for pages more than one level deep. YOu kind of always use `/static` and it should work for both local development (`manage.py runserver` and `localhost:8000/static`) and production environment in subfolder (`myserver.com/mytool/folder/page` and `myserver.com/mytool/static`)

Reply:
As discussed, done